app/models/tag.rb

```diff
@@ -45,16 +45,9 @@ def has_projects?
   end
 
   def notify_subscribers
-    repository.projects.without_versions.each do |project|
-      subscriptions = project.subscriptions
-      subscriptions = subscriptions.include_prereleases if prerelease?
-
-      subscriptions.group_by(&:notification_user).each do |user, _user_subscriptions|
-        next if user.nil?
-        next if user.muted?(project)
-        next if !user.emails_enabled?
-        VersionsMailer.new_version(user, project, self).deliver_later
-      end
+    project.mailing_list(include_prereleases: prerelease?).each do |user|
+      next if user.muted?(project)
```

Comment:
Aren't the project_mutes already checked in your updates in project.rb

Reply:
ah yes, I forgot I did that